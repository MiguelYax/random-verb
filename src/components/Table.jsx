import React from 'react';
import PropTypes from 'prop-types';
import Table from 'react-bootstrap/Table'

const CustomCard = ({ verb, past, pastParticipe, meaning, finalSound, darkMode }) => {
    return (

        <Table striped bordered hover variant={darkMode ? "dark" : "light"} >
            <thead>
                <tr>
                    <th colSpan="2">{meaning}</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td>Verb</td>
                    <td>{verb}</td>
                </tr>
                <tr>
                    <td>Past</td>
                    <td>{past}</td>
                </tr>
                <tr>
                    <td>Past Participe</td>
                    <td>{pastParticipe}</td>
                </tr>
                {finalSound &&
                    <tr>
                        <td>Final sound</td>
                        <td>{finalSound}</td>
                    </tr>
                }
            </tbody>
        </Table>

    )
}

CustomCard.propTypes = {
    verb: PropTypes.string,
    past: PropTypes.string,
    pastParticipe: PropTypes.string,
    meaning: PropTypes.string,
    finalSound: PropTypes.string,
    darkMode: PropTypes.bool,
}

export default CustomCard;
