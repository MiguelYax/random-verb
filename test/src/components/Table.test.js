import React from 'react';
import Table from '../../../src/components/Table';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const darkProps = {
    "verb": "Wake up",
    "past": "Woke up",
    "pastParticiple": "Woken up",
    "meaning": "Despertar",
    "finalSound": "",
    "darkMode": true
}

describe('When src/components/Table.jsx is used', () => {
    it('should render a table component', () => {
        const shallowTable = shallow(<Table {...darkProps} />)
        expect(shallowTable).toMatchSnapshot();
    });
    it('should render light mode and render final sound', () => {
        const lightProps = {
            ...darkProps,
            finalSound: "T",
            darkMode: false
        }
        const shallowTable = shallow(<Table {...lightProps} />)
        expect(shallowTable).toMatchSnapshot();
    });
});