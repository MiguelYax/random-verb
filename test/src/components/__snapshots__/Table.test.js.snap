// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When src/components/Table.jsx is used should render a table component 1`] = `
<ForwardRef
  bordered={true}
  className="text-light"
  hover={true}
  striped={true}
  variant="dark"
>
  <thead>
    <tr>
      <th
        colSpan="3"
      >
        Despertar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Verb
      </td>
      <td>
        Wake up
      </td>
      <td>
        <Speech
          text="Wake up"
        />
      </td>
    </tr>
    <tr>
      <td>
        Past
      </td>
      <td>
        Woke up
      </td>
      <td>
        <Speech
          text="Woke up"
        />
      </td>
    </tr>
    <tr>
      <td>
        Past participle
      </td>
      <td>
        Woken up
      </td>
      <td>
        <Speech
          text="Woken up"
        />
      </td>
    </tr>
    <tr>
      <td>
        Example
      </td>
      <td>
        I woke up when the rooster crowed.
      </td>
      <td>
        <Speech
          text="I woke up when the rooster crowed."
        />
      </td>
    </tr>
  </tbody>
</ForwardRef>
`;

exports[`When src/components/Table.jsx is used should render light mode and render final sound 1`] = `
<ForwardRef
  bordered={true}
  className="text-primary"
  hover={true}
  striped={true}
  variant="light"
>
  <thead>
    <tr>
      <th
        colSpan="3"
      >
        Despertar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Verb
      </td>
      <td>
        Wake up
      </td>
      <td>
        <Speech
          text="Wake up"
        />
      </td>
    </tr>
    <tr>
      <td>
        Past
      </td>
      <td>
        Woke up
      </td>
      <td>
        <Speech
          text="Woke up"
        />
      </td>
    </tr>
    <tr>
      <td>
        Past participle
      </td>
      <td>
        Woken up
      </td>
      <td>
        <Speech
          text="Woken up"
        />
      </td>
    </tr>
    <tr>
      <td>
        Final sound
      </td>
      <td
        colSpan="2"
      >
        T
      </td>
    </tr>
    <tr>
      <td>
        Example
      </td>
      <td>
        I woke up when the rooster crowed.
      </td>
      <td>
        <Speech
          text="I woke up when the rooster crowed."
        />
      </td>
    </tr>
  </tbody>
</ForwardRef>
`;
